src/Eloquent/Collection.php

```diff
@@ -49,4 +50,59 @@ public function toTree($childrenRelation = 'children')
 
         return $tree;
     }
+
+    /**
+     * Load parent/anscestor relations already present in the tree.
+     *
+     * @return static<int, TModel>
+     */
+    public function loadTreePathRelations()
+    {
+        $instance = $this->first();
+
+        if (is_null($instance)) {
+            return $this;
+        }
+
+        if (! method_exists($instance, 'getPathName') || ! method_exists($instance, 'getPathSeparator')) {
+            throw new RuntimeException(sprintf(
+                'Model [%s] does not have recusive relations.',
+                $instance::class,
+            ));
+        }
+
+        $keyName       = $instance->getKeyName();
+        $pathName      = $instance->getPathName();
+        $pathSeparator = $instance->getPathSeparator();
+
+        $lookup = $this->keyBy($keyName);
+
+        $keys = $this
+            ->pluck($pathName)
+            ->flatMap(fn (string $path): array => explode($pathSeparator, $path))
+            ->unique()
+            ->values();
+
+        $missing = $keys->diff($lookup->modelKeys());
+
+        if ($missing->isNotEmpty()) {
+            $lookup->merge($instance->newQuery()->findMany($missing)->keyBy($keyName));
+        }
+
+        foreach ($this->all() as $model) {
+            $path = array_reverse(explode($pathSeparator, $model->getAttribute($pathName)));
+
+            $ancestorsAndSelf = array_reduce(
+                $path,
+                fn ($collection, $step) => $collection->push($lookup[$step] ?? null),
+                $instance->newCollection(),
+            );
+
+            $model->setRelation('parent', count($path) > 1 ? $lookup[$path[1]] : null);
+            $model->setRelation('ancestorsAndSelf', $ancestorsAndSelf);
+            $model->setRelation('ancestors', $ancestorsAndSelf->slice(0, -1));
```

Comment:
```suggestion
            $model->setRelation('ancestors', $ancestorsAndSelf->slice(1));
```

This fixes the issue spotted by @staudenmeir